tools/online_editor/src/editor_widget.ts

```diff
@@ -579,7 +621,7 @@ export class EditorWidget extends Widget {
     ];
   }
 
-  set_demo(location: string) {
+  async set_demo(location: string) {
```

Comment:
The call site is this:

```typescript
   commands.addCommand(command_name, {
      label: demo[1],
      execute: () => {
        editor.set_demo(demo[0]);
      },
    });
```

The `ICommandOptions` that `addCommand` takes specifies that execute can also return a promise (`execute: CommandFunc<any | Promise<any>>;`) 

I think it would be cleanest to then also return the promise of `set_demo`, i.e. writing `return editor.set_demo(demo[0]));`.

That said, that's a little synthetic - as the luminous call site just consumes - but in theory somebody could call `execute()` in the `CommandRegistry` and expect the correct promise.

Reply:
Sorry, I had that in a local change but forgot to add it here:-)